src/middlewared/middlewared/plugins/netdata.py

```diff
@@ -0,0 +1,230 @@
+        if bind_to_ips:
+            valid_ips = [ip['address'] for ip in await self.middleware.call('interfaces.ip_in_use')]
+            valid_ips.extend(['127.0.0.1', '::1', '*'])
+
+            for bind_ip in bind_to_ips:
+                if bind_ip not in valid_ips:
+                    verrors.add(
+                        'netdata_update.bind_to',
+                        f'Invalid {bind_ip} bind to IP'
+                    )
+        else:
+            verrors.add(
+                'netdata_update.bind_to',
+                'This field is required'
+            )
+
+        update_alarms = data.pop('update_alarms', {})
+        valid_alarms = await self.list_alarms()
+        if update_alarms:
+            for alarm in update_alarms:
+                if alarm not in valid_alarms:
+                    verrors.add(
+                        'netdata_update.alarms',
+                        f'{alarm} not a valid alarm'
+                    )
+
+            verrors.extend(
+                validate_attributes([Bool(key) for key in update_alarms], {'attributes': update_alarms})
+            )
+
+        # Validating streaming metrics now
+        stream_mode = data.get('stream_mode')
+        if stream_mode == 'SLAVE':
+            for key in ('api_key', 'destination'):
+                if not data.get(key):
+                    verrors.add(
+                        f'netdata_update.{key}',
+                        f'{key} is required with stream mode as SLAVE'
+                    )
+        elif stream_mode == 'MASTER':
+            for key in ('allow_from', 'api_key'):
+                if not data.get(key):
+                    verrors.add(
+                        f'netdata_update.{key}',
+                        f'{key} is required with stream mode as MASTER'
+                    )
+
+        verrors.check()
+
+        for alarm in valid_alarms:
+            # Let's add alarms to our db if they aren't already there
+            if alarm not in data['alarms']:
+                data['alarms'][alarm] = True
+
+        for alarm in update_alarms:
+            # These are valid alarms
+            data['alarms'][alarm] = update_alarms[alarm]
+
+        data['memory_mode'] = data['memory_mode'].lower()
+        data['stream_mode'] = data['stream_mode'].lower()
+
+        return data
+
+    @accepts(
+        Dict(
+            'netdata_update',
+            Str('additional_params'),
+            Dict(
+                'alarms',
+                additional_attrs=True
+            ),
+            List('allow_from', items=[Str('pattern')]),
+            Str('api_key', validators=[UUID()]),
+            List('bind_to', items=[Str('bind_to_ip')]),
+            Int('bind_to_port', validators=[Port()]),
+            List('destination', items=[Str('destination', validators=[IpAddress(port=True)])]),
+            Int('history'),
+            Int('http_port_listen_backlog'),
+            Str('memory_mode', enum=['SAVE', 'MAP', 'RAM', 'NONE']),
```

Comment:
If SAVE, where exactly is the data stored on disk?

Reply:
WIth memory mode set to save netdata saves it's round robin db on exit. Normally for it's operations it uses `/var/db/netdata` and I think it saves this there as well

Reply:
If we are to support SAVE this will need to be fixed. /var/db/netdata is a memory filesystem, lost on reboot.